sdk/src/main/scala/com/horizen/certificatesubmitter/strategies/WithoutKeyRotationStrategy.scala

```diff
@@ -0,0 +1,122 @@
+
+    val btrFee: Long = getBtrFee(status.referencedEpoch)
+    val consensusEpochNumber = lastConsensusEpochNumberForWithdrawalEpochNumber(history, status.referencedEpoch)
+    val ftMinAmount: Long = getFtMinAmount(consensusEpochNumber)
+    val endEpochCumCommTreeHash = lastMainchainBlockCumulativeCommTreeHashForWithdrawalEpochNumber(history, status.referencedEpoch)
+    val sidechainId = params.sidechainId
+    val utxoMerkleTreeRoot: Seq[Array[Byte]] = getUtxoMerkleTreeRoot(status.referencedEpoch, state)
+
+
+    val signersPublicKeyWithSignatures = params.signersPublicKeys.zipWithIndex.map {
+      case (pubKey, pubKeyIndex) =>
+        (pubKey, status.knownSigs.find(info => info.pubKeyIndex == pubKeyIndex).map(_.signature))
+    }
+
+    DataForProofGenerationWithoutKeyRotation(
+      status.referencedEpoch,
+      sidechainId,
+      withdrawalRequests,
+      endEpochCumCommTreeHash,
+      btrFee,
+      ftMinAmount,
+      utxoMerkleTreeRoot,
+      signersPublicKeyWithSignatures)
+  }
+
+  override def getMessageToSign(referencedWithdrawalEpochNumber: Int): Try[Array[Byte]] = Try {
+    def getMessage(sidechainNodeView: View): Try[Array[Byte]] = Try {
+      val history = sidechainNodeView.history
+      val state = sidechainNodeView.state
+
+      val withdrawalRequests: Seq[WithdrawalRequestBox] = state.withdrawalRequests(referencedWithdrawalEpochNumber)
+
+      val btrFee: Long = getBtrFee(referencedWithdrawalEpochNumber)
+      val consensusEpochNumber = lastConsensusEpochNumberForWithdrawalEpochNumber(history, referencedWithdrawalEpochNumber)
+      val ftMinAmount: Long = getFtMinAmount(consensusEpochNumber)
+
+      val endEpochCumCommTreeHash = lastMainchainBlockCumulativeCommTreeHashForWithdrawalEpochNumber(history, referencedWithdrawalEpochNumber)
+      val sidechainId = params.sidechainId
+
+      val utxoMerkleTreeRoot: Seq[Array[Byte]] = {
+        Try {
+          getUtxoMerkleTreeRoot(referencedWithdrawalEpochNumber, state)
+        } match {
+          case Failure(e: IllegalStateException) =>
+            throw new Exception("CertificateSubmitter is too late against the State. " +
+              s"No utxo merkle tree root for requested epoch $referencedWithdrawalEpochNumber. " +
+              s"Current epoch is ${state.getWithdrawalEpochInfo.epoch}")
+          case Failure(exception) => log.error("Exception while getting utxoMerkleTreeRoot", exception)
+            throw new Exception(exception)
+          case Success(value) => value
+        }
+      }
+
+      CryptoLibProvider.sigProofThresholdCircuitFunctions.generateMessageToBeSigned(
+        withdrawalRequests.asJava,
+        sidechainId,
+        referencedWithdrawalEpochNumber,
+        endEpochCumCommTreeHash,
+        btrFee,
+        ftMinAmount,
+        utxoMerkleTreeRoot)
+    }
+
+    Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView(getMessage), timeoutDuration).asInstanceOf[Try[Array[Byte]]].get
```

Comment:
Strategy should not care about `sidechainNodeViewHolderRef` and actor messages.
It is a work for Certificate Submitter.
Instead, we may pass `sidechainNodeView: View` as an argument of `getMessageToSign`:
`override def getMessageToSign(sidechainNodeView: View, referencedWithdrawalEpochNumber: Int): Try[Array[Byte]]`